src/main/java/de/bonndan/nivio/output/icons/LocalIcons.java

```diff
@@ -0,0 +1,115 @@
+package de.bonndan.nivio.output.icons;
+
+import de.bonndan.nivio.model.Item;
+import de.bonndan.nivio.model.Label;
+import de.bonndan.nivio.util.URLHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import java.net.URL;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static de.bonndan.nivio.output.icons.IconMapping.DEFAULT_ICON;
+
+/**
+ * Provides icons and vendor icons as data.
+ *
+ *
+ */
+@Service
+public class LocalIcons {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LocalIcons.class);
+
+    private final VendorIcons vendorIcons;
+
+    /**
+     * default icon data url
+     */
+    private final String defaultIcon;
+
+    /**
+     * data url cache
+     */
+    private final Map<String, String> iconDataUrls = new ConcurrentHashMap<>();
+
+
+    public LocalIcons(VendorIcons vendorIcons) {
+
+        this.vendorIcons = vendorIcons;
+        defaultIcon = getIconUrl(DEFAULT_ICON.getIcon(), false);
+    }
+
+    public String getIconUrl(Item item) {
+
+        //icon label based
+        String icon = item.getIcon();
+        if (!StringUtils.isEmpty(icon)) {
+
+            if (icon.startsWith(VendorIcons.VENDOR_PREFIX)) {
+                String key = icon.replace(VendorIcons.VENDOR_PREFIX, "").toLowerCase();
+                return vendorIcons.getUrl(key).orElse(defaultIcon);
```

Comment:
This is the only place where `vendorIcons.getUrl()` is used, so I was at first a little confused about what `LocalIcons` really is. I thought `LocalIcons` are icons addes by the user? Is this correct? (In this case, maybe a different naming would be better. Why not just Icons?)

Or does it mean that there are not retrieved from a remote location such as the vendor icons?

Reply:
LocalIcons is a crappy name, but I had no better idea. It stands for the locally (fs) available svg icons that come shipped with nivio. They are loaded and embedded, so nivio does not serve anything anymore.

Reply:
Okay, I think for the users it is okay. As I developer, I now understand. Maybe we can add a brief comment to the class, so others developers understand too?

In this case I also did not come up with a better name. "BuiltInIcons" could be another idea.

Reply:
One more ugly aspect is that it is named to handle local icons, but suddenly comes up with remote vendor icons. I don't like my design here.

Reply:
Yes, that is exactly what I was wondering about.

Reply:
refactored.